component/attr_cache/attr_cache.go

```diff
@@ -505,6 +505,19 @@ func (ac *AttrCache) CreateLink(options internal.CreateLinkOptions) error {
 	return err
 }
 
+// FlushFile : flush file
+func (ac *AttrCache) FlushFile(options internal.FlushFileOptions) error {
+	log.Trace("AttrCache::FlushFile : %s", options.Handle.Path)
```

Comment:
In file-cache case, flushFile was not being forwarded as flush was resulting into uploadFromFile and that was taking care of invalidating data. Are we invalidating here because stream layer forwards the flushFile call ?
Also, if so we shall have code in streaming layer that if close is called without  a flush call and fir handle is dirty then implicitly call flush on stream layer (same is done in file-cache as well)

Reply:
yes, we are invalidating here because stream layer forwards to flush call. How can we repro a case of close without flush? I was not able to do that. For example, in Python if you close a handle flush is always called first on the kernel, even if we just read without writing.
If no flush call comes then we should not flush.

Reply:
yes, application can do a close without flush however implicitly its converted to flush and close. Just to ensure on some system in case we directly get a close, in file-cache we have handled this explicitly. I remember there was a test case where we were able to simulate this. If you do a scp from outside to directly a mounted directory, I guess. Not able to recollect the exact case but there was one condition in which we were able to hit this, and it was fixed in v1 as well. So better to have that check in streaming as well.

Reply:
done.